### PR TITLE
fix(TCOMP-2475): Migration from higher version not protected for a

### PR DESCRIPTION
**Pull request by yyin-talend**
https://jira.talendforge.org/browse/TCOMP-2475

connector

### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

**Review by undx — COMMENTED** [review]
LGTM
See with Axel for api-tester IT

**Comment by acatoire** [comment]
IT test are now Green on this test, so the ticket is ready for me. I did the validation on the branch.

File: component-server-parent/component-server/src/test/java/org/talend/sdk/component/server/front/ComponentResourceImplTest.java [hunk 1]
```diff
@@ -179,6 +178,23 @@ void migrate() {
         assertEquals("true", migrated.get("migrated"));
     }
 
+    @Test
+    void no_migrate() {
```
Comment (undx) [inline component-server-parent/component-server/src/test/java/org/talend/sdk/component/server/front/ComponentResourceImplTest.java:182]: Please avoid underscore in method's name.
```suggestion
    void protectHigherMigration() {
```
or 
```suggestion
    void noMigration() {
```

**Comment by undx** [comment]
> IT tests are now Green on this test, so the ticket is ready for me. I did the validation on the branch. Regarding the Unit test, we have two cases (if (version > and if version ==) to have a complete coverage of branches could you add the other unit test?

There's already an UT testing the case. But, yes we can add more tests on subject @yyin-talend.

**Review by undx — APPROVED** [review]
lgtm

**Pull request merged**